src/game/audio.rs

```diff
@@ -0,0 +1,154 @@
+use bevy::prelude::*;
+use bevy_kira_audio::{Audio, AudioChannel, AudioPlugin, AudioSource};
+use rand::{thread_rng, Rng};
+
+use crate::AppState;
+
+pub struct GameAudioPlugin;
+
+pub struct AudioHitEvent;
+pub struct AudioCoffeeEvent;
+pub struct AudioRustEvent;
+pub struct AudioFastShootEvent;
+pub struct AudioShootEvent;
+pub struct AudioDeadPlayerEvent;
+
+pub struct AudioAssets {
+    bg: Handle<AudioSource>,
+    menu: Handle<AudioSource>,
+    hit1: Handle<AudioSource>,
+    hit2: Handle<AudioSource>,
+    hit3: Handle<AudioSource>,
+    death: Handle<AudioSource>,
+    drink1: Handle<AudioSource>,
+    drink2: Handle<AudioSource>,
+    drink3: Handle<AudioSource>,
+    shoot: Handle<AudioSource>,
+    fast_shoot: Handle<AudioSource>,
+    lvlup: Handle<AudioSource>,
+    bg_channel: AudioChannel,
+    menu_channel: AudioChannel,
+}
+
+impl Plugin for GameAudioPlugin {
+    fn build(&self, app: &mut App) {
+        app.add_plugin(AudioPlugin)
+            .add_startup_system_to_stage(StartupStage::PreStartup, load_audio)
+            .add_system_set(SystemSet::on_enter(AppState::InGame).with_system(play_bg_music))
+            .add_system_set(SystemSet::on_exit(AppState::InGame).with_system(play_menu_music))
+            .add_system(play_hit_sfx)
+            .add_system(play_death_sfx)
+            .add_system(play_eat_sfx)
+            .add_system(play_lvlup_sfx)
+            .add_system(play_shoot_sfx)
+            .add_system(play_fast_shoot_sfx)
+            .add_startup_system(play_menu_music)
+            .add_event::<AudioRustEvent>()
+            .add_event::<AudioCoffeeEvent>()
+            .add_event::<AudioHitEvent>()
+            .add_event::<AudioFastShootEvent>()
+            .add_event::<AudioShootEvent>()
+            .add_event::<AudioDeadPlayerEvent>();
+    }
+}
+
+pub fn play_hit_sfx(
+    audio: Res<Audio>,
+    audio_state: Res<AudioAssets>,
+    mut audio_event: EventReader<AudioHitEvent>,
+) {
+    audio_event.iter().for_each(|_| {
+        let mut rng = thread_rng();
+        match rng.gen_range(0..100) {
```

Comment:
a to nie prościej zrobić losowanie od 1 do 3? (czy tam od 0 do 2)

Reply:
Zdecydowanie prościej...